tests/backup/backup_test.go

```diff
@@ -3934,6 +3935,180 @@ var _ = Describe("{ResizeOnRestoredVolume}", func() {
+						bucketName, orgID, "")
+				}
+			}
+			log.InfoD("Successfully created locked buckets and backup location")
+		})
+
+		Step("Register cluster for backup", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			CreateSourceAndDestClusters(orgID, "", "", ctx)
+			clusterStatus, clusterUid = Inst().Backup.RegisterBackupCluster(orgID, SourceClusterName, "")
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, "Verifying backup cluster")
+		})
+
+		for _, namespace := range bkpNamespaces {
+			for backupLocationUID, backupLocationName := range BackupLocationMap {
+				Step("Taking backup of application to locked bucket", func() {
+					ctx, err := backup.GetAdminCtxFromSecret()
+					dash.VerifyFatal(err, nil, "Getting context")
+					preRuleUid, _ := Inst().Backup.GetRuleUid(orgID, ctx, preRuleNameList[0])
+					postRuleUid, _ := Inst().Backup.GetRuleUid(orgID, ctx, postRuleNameList[0])
+					backupName := fmt.Sprintf("%s-%s-%s", BackupNamePrefix, namespace, backupLocationName)
+					backupList = append(backupList, backupName)
+					CreateBackup(backupName, SourceClusterName, backupLocationName, backupLocationUID, []string{namespace},
+						labelSelectors, orgID, clusterUid, preRuleNameList[0], preRuleUid, postRuleNameList[0], postRuleUid, ctx)
+				})
+				Step("Restoring the backups application", func() {
+					ctx, err := backup.GetAdminCtxFromSecret()
+					log.FailOnError(err, "Fetching px-central-admin ctx")
+					CreateRestore(fmt.Sprintf("%s-restore", backupName), backupName, nil, SourceClusterName, orgID, ctx)
+				})
+				Step("Resize volume after the restore is completed", func() {
+					log.InfoD("Resize volume after the restore is completed")
+					var err error
+					for _, ctx := range contexts {
+						var appVolumes []*volume.Volume
+						log.InfoD(fmt.Sprintf("get volumes for %s app", ctx.App.Key))
+						appVolumes, err = Inst().S.GetVolumes(ctx)
+						log.FailOnError(err, "Failed to get volumes for app %s", ctx.App.Key)
+						dash.VerifyFatal(len(appVolumes) > 0, true, "App volumes exist?")
+						var requestedVols []*volume.Volume
+						log.InfoD(fmt.Sprintf("Increase volume size %s on app %s's volumes: %v",
+							Inst().V.String(), ctx.App.Key, appVolumes))
+							requestedVols, err = Inst().S.ResizeVolume(ctx, Inst().ConfigMap)
+							log.FailOnError(err, "Volume resize successful ?")
+							log.InfoD(fmt.Sprintf("validate successful volume size increase on app %s's volumes: %v",
+								ctx.App.Key, appVolumes))
+								for _, v := range requestedVols {
+									// Need to pass token before validating volume
+									params := make(map[string]string)
+									if Inst().ConfigMap != "" {
+										params["auth-token"], err = Inst().S.GetTokenFromConfigMap(Inst().ConfigMap)
+										log.FailOnError(err, "Failed to get token from configMap")
+									}
+									err := Inst().V.ValidateUpdateVolume(v, params)
```

Comment:
Helper to check if the volume what application is using is really expanded by execing into pod and checking the volume size

Reply:
Done